nautobot/core/tests/test_csv.py

```diff
@@ -0,0 +1,286 @@
+import re
+
+from django.test import RequestFactory, TestCase, override_settings
+from django.urls import reverse
+from nautobot.core.constants import CSV_NON_TYPE, CSV_OBJECT_NOT_FOUND
+
+from nautobot.dcim.api.serializers import DeviceSerializer
+from nautobot.dcim.models.devices import Device, DeviceType
+from nautobot.dcim.models.locations import Location
+from nautobot.extras.models.roles import Role
+from nautobot.extras.models.statuses import Status
+from nautobot.extras.models.tags import Tag
+from nautobot.tenancy.models import Tenant
+from nautobot.users.factory import UserFactory
+
+
+class CSVParsingRelatedTestCase(TestCase):
```

Comment:
This Test covers all use case I can think of:

- FK field with some missing `natural_key_lookup` values i.e `ObjectNotFound`
- FK field = None i.e all FK field lookups are `ObjectNotFound`
- FK fields without any missing lookup value